factory/django.py

```diff
@@ -151,7 +152,19 @@ def _get_or_create(cls, model_class, *args, **kwargs):
             key_fields[field] = kwargs.pop(field)
         key_fields['defaults'] = kwargs
 
-        instance, _created = manager.get_or_create(*args, **key_fields)
+        try:
+            instance, _created = manager.get_or_create(*args, **key_fields)
+        except IntegrityError:
+            try:
+                instance = manager.get(**cls.original_kwargs)
+            except manager.model.DoesNotExist:
+                raise ValueError(
+                    "django_get_or_create - Unable to create a new object "
+                    "due an IntegrityError raised based on "
+                    "your model's uniqueness constraints. "
+                    "DoesNotExist: Unable to find an existing object based on "
```

Comment:
To avoid chaining in the error message, could we use exception chaining for Python3? https://www.python.org/dev/peps/pep-3134/

Reply:
Great idea! We still have to support Python 2.7 for a while though, and I am not convinced it's worth having two different code paths for Python 2 and 3. Are there any standard solutions for this issue?

Reply:
There seems to be a [`six.raise_from`](https://pythonhosted.org/six/#six.raise_from). `six` is not required by factory_boy though, and I don't think it should be introduced when Python 2 will die in a few months.

Two code paths is the easiest approach I can think of. Otherwise, this could be documented with a TODO.